gammapy/detect/cwt.py

```diff
@@ -108,34 +139,48 @@ class CWT(object):
         - Fill ``self.model += res * (res > 0)``
         - What is this??
 
-
     - max_scale_image : 2D image, value = index of scale where emission is most significant
 
-    Parameters
+    Examples
+    --------
+    TODO
+
+    References
     ----------
-    min_scale : float
-        first scale used
-    nscales : int
-        number of scales considered
-    scale_step : float
-        base scaling factor
+    R. Terrier, L. Demanet, I.A. Grenier, and J.P. Antoine. Wavelet
+    analysis of EGRET data. Universite   Paris VII & Sap CEA Saclay.
+    Institut de physique theorique et mathematique, Universite
+    catholique de Louvain, Louvain-la-Neuve
     """
 
-    def __init__(self, min_scale, nscales, scale_step):
+    def __init__(self, min_scale, nscales, scale_step, max_iter=10, tol=1e-5):
         self.kernbase = dict()
+
         self.scales = dict()
         self.nscale = nscales
         self.scale_step = scale_step
+
         for idx_scale in np.arange(0, nscales):
             scale = min_scale * scale_step ** idx_scale
             self.scales[idx_scale] = scale
             self.kernbase[idx_scale] = difference_of_gauss_kernel(scale, scale_step)
+        self.max_iter=max_iter
+        self.tol=tol
 
         max_scale = min_scale * scale_step ** nscales
         self.kern_approx = gauss_kernel(max_scale)
 
     def set_data(self, counts, background):
-        """Set input images."""
+        """
+        Set input images.
+
+        TODO : RENAME.
+        Like fit methods in sklearn. I like that.
```

Comment:
One option to handle the data could be a separate class:
```
class CWTData:
    def __init__(self, counts, background, ...):
       self.counts = counts
       self.background = background

```
and then to use it in the algorithm class:
```
class CWT:
    def __init__(self, data, parameters):
        self.data = data
```
i.e. collect all the images in a data object.

And a user would use it like this:
```
from gammapy.detect import CWT, CWTData

# read input data and pass it to the data container class
data = CWTData(...)
cwt = CWT(data, parameters)
cwt.run()
```

You could also just use a http://docs.gammapy.org/en/latest/api/gammapy.image.SkyImageCollection.html or an OrderedDict instead of inventing a `CWTData`.

There's many ways to structure such code / algorithms and we don't have established patterns yet in Gammapy.

@OlgaVorokh - Please think about it and make a proposal what your preference would be for discussion before implementing it.